src/libs/actions/Policy.ts

```diff
@@ -440,7 +440,7 @@ function setWorkspaceAutoReporting(policyID: string, enabled: boolean, frequency
             value: {
                 autoReporting: enabled,
                 harvesting: {
-                    enabled: true,
+                    enabled,
                 },
                 autoReportingFrequency: frequency,
```

Comment:
In case of failure we need to revert the 3 values (currently only autoReporting is reverted)

Reply:
Those other functions should be fixed as well. If any request fail the user should have visual indication that it did fail by reverting to the original value

Reply:
@DylanDylann s77rt is right, we need to update the failure case everywhere here seems like, do you need help figuring out the correct state? 

cc @lakchote @luacmartins for visibility, we missed this in the OG review